Client/Frontend/Browser/PageZoom/PageZoomSettingsView.swift

```diff
@@ -0,0 +1,170 @@
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+import SwiftUI
+import Shared
+import BraveShared
+import Data
+import CoreData
+
+extension Domain: Identifiable {
+  public var id: String {
+    url ?? UUID().uuidString
+  }
+}
+
+private extension Domain {
+  var host: String {
+    URL(string: url ?? "")?.host ?? ""
+  }
+}
+
+struct PageZoomSettingsView: View {
+  @Environment(\.managedObjectContext) private var context
+  
+  @State private var domains = PageZoomSettingsView.filteredDomains(DataController.swiftUIContext)
+  
+  @State private var defaultZoomLevel = Preferences.General.defaultPageZoomLevel.value
+  
+  var body: some View {
+    List {
+      if !domains.isEmpty {
+        Section(
+          header: Text(Strings.PageZoom.specificWebsitesZoomLevelsSectionTitle)
+            .font(.footnote)
+            .foregroundColor(Color(UIColor.secondaryBraveLabel))
+        ) {
+          ForEach(domains) { domain in
+            HStack {
+              let url = domain.url ?? ""
+              
+              let zoomLevel = domain.zoom_level?.doubleValue ?? 1.0
+              
+              let percentageLevel = Int(zoomLevel * 100.0)
+              
+              Text(url)
+                .font(.headline)
+                .foregroundColor(Color(UIColor.braveLabel))
+              
+              Spacer()
+              
+              Text(PageZoomView.percentFormatter.string(from: .init(value: zoomLevel)) ?? "\(percentageLevel)%")
+                .foregroundColor(Color(UIColor.braveLabel))
+                .multilineTextAlignment(.trailing)
+            }
+          }
+          .onDelete(perform: resetItemZoomLevel)
+        }
+        .listRowBackground(Color(UIColor.secondaryBraveGroupedBackground))
+      }
+      
+      Section(
+        header: Text(domains.isEmpty ? Strings.PageZoom.defaultZoomLevelSectionTitle : Strings.PageZoom.otherWebsiteZoomLevelSectionTitle)
+          .font(.footnote)
+          .foregroundColor(Color(UIColor.secondaryBraveLabel))
+      ) {
+        ForEach(PageZoomView.steps, id: \.self) { step in
+          
+          let zoomLevel = Int(step * 100.0)
+          let preferredZoomLevel = Int(defaultZoomLevel * 100.0)
+         
+          Button(action: {
+            Preferences.General.defaultPageZoomLevel.value = step
+            defaultZoomLevel = step
+            
+            domains = PageZoomSettingsView.filteredDomains(context)
+          }, label: {
+            HStack(spacing: 0.0) {
+              Text(PageZoomView.percentFormatter.string(from: .init(value: step)) ?? "\(zoomLevel)%")
+                .foregroundColor(Color(UIColor.braveLabel))
+                .multilineTextAlignment(.trailing)
+              
+              Spacer()
+              
+              if preferredZoomLevel == zoomLevel {
+                Image(systemName: "checkmark")
```

Comment:
Probably needs a medium/semibold font weight to match standard table cell style checkmark